var/spack/repos/builtin/packages/py-scikit-learn/package.py

```diff
@@ -21,12 +40,50 @@ class PyScikitLearn(PythonPackage):
     version('0.13.1', 'acba398e1d46274b8470f40d0926e6a4')
 
-    depends_on('python@2.6:2.8,3.3:', when='@:0.19.1')
-    depends_on('python@2.7:2.8,3.4:', when='@0.20.0:')
-    depends_on('py-numpy@1.6.1:', type=('build', 'run'), when='@:0.19.1')
-    depends_on('py-numpy@1.8.2:', type=('build', 'run'), when='@0.20.0:')
-    depends_on('py-scipy@0.9:', type=('build', 'run'), when='@:0.19.1')
-    depends_on('py-scipy@0.13.3:', type=('build', 'run'), when='@0.20.0:')
+    depends_on('python@2.6:2.8,3.3:', when='@:0.19')
+    depends_on('python@2.7:2.8,3.4:', when='@0.20.0:0.20.999')
+    depends_on('python@3.5:', when='@0.21:')
+    depends_on('py-numpy@1.6.1:', type=('build', 'run'), when='@:0.19')
+    depends_on('py-numpy@1.8.2:', type=('build', 'run'), when='@0.20.0:0.20.999')
+    depends_on('py-numpy@1.11.0:', type=('build', 'run'), when='@0.21:')
+    depends_on('py-scipy@0.9:', type=('build', 'run'), when='@:0.19')
+    depends_on('py-scipy@0.13.3:', type=('build', 'run'), when='@0.20.0:0.20.999')
+    depends_on('py-scipy@0.17.0:', type=('build', 'run'), when='@0.21:')
+    depends_on('py-joblib@0.11:', type=('build', 'run'))
     depends_on('py-cython@0.23:', type='build')
-    depends_on('py-test@3.3.0:', type='test')
+    depends_on('py-cython@0.28.5:', type='build', when='@0.21:')
+    depends_on('py-pytest@3.3.0:', type='test')
+    depends_on('py-pandas', type='test')
     depends_on('py-setuptools', type='build')
+    depends_on('ompt-openmp', when='%clang platform=darwin')
+
+    def setup_environment(self, spack_env, run_env):
+        # https://scikit-learn.org/stable/developers/advanced_installation.html#mac-osx
+        if self.spec.satisfies('%clang platform=darwin'):
+            spack_env.append_flags(
+                'CPPFLAGS', '-Xpreprocessor -fopenmp')
+            spack_env.append_flags(
+                'CFLAGS', self.spec['ompt-openmp'].headers.include_flags)
+            spack_env.append_flags(
+                'CXXFLAGS', self.spec['ompt-openmp'].headers.include_flags)
+            spack_env.append_flags(
+                'LDFLAGS', self.spec['ompt-openmp'].libs.ld_flags)
+            spack_env.append_flags(
+                'DYLD_LIBRARY_PATH',
+                self.spec['ompt-openmp'].libs.directories[0])
+
+            run_env.append_flags(
+                'DYLD_LIBRARY_PATH',
+                self.spec['ompt-openmp'].libs.directories[0])
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        if self.spec.satisfies('%clang platform=darwin'):
+            spack_env.append_flags(
+                'DYLD_LIBRARY_PATH',
+                self.spec['ompt-openmp'].libs.directories[0])
+
+    # def install_test(self):
+    #     # https://scikit-learn.org/stable/developers/advanced_installation.html#testing
+    #     with working_dir('spack-test', create=True):
+    #         pytest = which('pytest')
+    #         pytest('sklearn')
```

Comment:
I followed the testing instructions, but it told me that the `sklearn` directory doesn't exist. I'm not sure how this was supposed to work...